app/src/main/kotlin/com/simplemobiletools/notes/pro/helpers/NotesExporter.kt

```diff
@@ -0,0 +1,52 @@
+package com.simplemobiletools.notes.pro.helpers
+
+import android.content.Context
+import com.google.gson.Gson
+import com.google.gson.stream.JsonWriter
+import com.simplemobiletools.commons.helpers.PROTECTION_NONE
+import com.simplemobiletools.commons.helpers.ensureBackgroundThread
+import com.simplemobiletools.notes.pro.extensions.notesDB
+import com.simplemobiletools.notes.pro.models.Note
+import java.io.OutputStream
+
+class NotesExporter(private val context: Context) {
+    enum class ExportResult {
+        EXPORT_FAIL, EXPORT_OK
+    }
+
+    private val gson = Gson()
+
+    fun exportNotes(outputStream: OutputStream?, onProgress: (total: Int, current: Int) -> Unit = { _, _ -> }, callback: (result: ExportResult) -> Unit) {
```

Comment:
onProgress is unused, right? Thats how it is in SMS Messenger too. Exporting messages is quick, I dont think we will ever need a progress bar for it.